### PR TITLE
Add all lz4 sources

### LINKED ISSUE #45: undefined symbol: LZ4F_createDecompressionContext since MCAP 0.1.2 upgrade
## Description
* https://github.com/foxglove/mcap/pull/482 , which was included in v0.1.2, added a new dependency on `lz4frames.h` (and its implementing `.c`)
*  https://github.com/ros-tooling/rosbag2_storage_mcap/pull/42 upgraded to this new version without incorporating the new sources, so we get an undefined symbol error when actually trying to read/play mcap bags

This is luckily not synced into Humble yet, but the latest bloomed version 0.1.6 includes this change, which will go out in the next sync.

I'm fixing it right now, just wanted to document the issue while I do so

### PR DESCRIPTION
**Pull request by emersonknapp**
We specifically to need `lz4frame.c` for `LZ4F_createDecompressionContext` etc. Do as a glob to futureproof against mcap including other potential source files.

Fixes #45

### CODE REVIEW

**Comment by emersonknapp** [comment]
Sorry for the verbosity of the test file - it was a little tricky to make it simultaneously compatible with Foxy+Galactic+Humble(+Rolling). Had to fall back to very low level APIs, since the convenient short ones were added in a "usable C++ API" push that we did after Foxy.

File: rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    writer.open(options, rosbag2_cpp::ConverterOptions{});
+    writer.create_topic(topic_metadata);
+
+    auto serialized_msg = std::make_shared<rclcpp::SerializedMessage>();
+    serialization.serialize_message(&msg, serialized_msg.get());
+
+    auto serialized_bag_msg = std::make_shared<rosbag2_storage::SerializedBagMessage>();
+    serialized_bag_msg->serialized_data = std::shared_ptr<rcutils_uint8_array_t>(
+      const_cast<rcutils_uint8_array_t*>(&serialized_msg->get_rcl_serialized_message()),
+      [](rcutils_uint8_array_t* /* data */) {});
```
Comment (jtbandes) [inline rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp:81]: Seems like a misuse of shared_ptr, so probably could at least use a comment explaining why it has to be this way

**Review by jtbandes — APPROVED** [review]
This code looks extremely sketchy but it's all in the tests, so I'm not too concerned 😀

File: rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp (hunk 1 above)
Comment (emersonknapp) [reply rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp:81]: agreed - it's a big misuse of everything. added a comment. when Foxy hits EOL we can really simplify this test.

**Pull request merged**